state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -467,34 +508,30 @@ pub mod pallet {
 			amount: Option<AssetAmount>,
 		},
 		SwapAmountConfiscated {
-			swap_id: SwapId,
-			source_asset: Asset,
-			destination_asset: Asset,
+			swap_request_id: SwapRequestId,
+			asset: Asset,
 			total_amount: AssetAmount,
 			confiscated_amount: AssetAmount,
 		},
 		SwapEgressIgnored {
-			swap_id: SwapId,
+			swap_request_id: SwapRequestId,
 			asset: Asset,
 			amount: AssetAmount,
 			reason: DispatchError,
 		},
 		RefundEgressIgnored {
-			swap_id: SwapId,
+			swap_request_id: SwapRequestId,
 			asset: Asset,
 			amount: AssetAmount,
 			reason: DispatchError,
 		},
-		NetworkFeeTaken {
-			swap_id: SwapId,
-			fee_amount: AssetAmount,
-		},
 		BuyIntervalSet {
 			buy_interval: BlockNumberFor<T>,
 		},
 		SwapRetryDelaySet {
 			swap_retry_delay: BlockNumberFor<T>,
 		},
+		// TODO: add SwapFailed?
```

Comment:
I left it as a todo since we haven't quite agreed on how this should work. For example, do we emit it for every swap in a batch every time the batch fails even it will be retried (seems a bit excessive), or do we only emit when we "give up" on the swap and refund it (which means it will never be emitted for non-FoK swaps)?

IMO it is OK to add this in a future PR (while we get some clarity on it) since this event it "new", i.e. it doesn't replace any existing one that the product is expecting.

Reply:
Ok, is there a ticket to follow up on this? Is this for the product specifically or?

Reply:
Started a thread on discord.